File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0, r = 1, sol = 0;
        while (r < prices.size()) {
            if (prices[l] < prices[r]) {
                int profit = prices[r]-prices[l];
                sol = max(sol, profit);
            } else {
                l = r;
            }
            r++;
        }
        return sol;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        HashSet<int> in_window = new();
        int l = 0, r = 0, sol = 0;
        while (r < s.Length) {
            if (!in_window.Contains(s[r])) {
                in_window.Add(s[r]);
                r++;
            } else {
                sol = Math.Max(sol, r-l);
                while (in_window.Contains(s[r])) {
                    in_window.Remove(s[l]);
                    l++;
                }
            }
        }
        sol = Math.Max(sol, r-l);
        return sol;
    }
}
